// @shopify/cli/dist/cli/commands/demo/index.js
import {
  demo
} from "../../../chunk-ZOTASNQ2.js";
import "../../../chunk-RP2ISWQN.js";
import "../../../chunk-LAY4AEXX.js";
import {
  base_command_default
} from "../../../chunk-EDVQGTOL.js";
import "../../../chunk-4XNS74OG.js";
import "../../../chunk-DWXXFI3U.js";
import "../../../chunk-CKKROMZF.js";
import "../../../chunk-LKGDG6WW.js";
import "../../../chunk-63QEL3SZ.js";
import "../../../chunk-QX3YP6O5.js";
import "../../../chunk-HTYJDJRZ.js";
import "../../../chunk-B4KDQK3X.js";
import "../../../chunk-OQBI57X3.js";
import "../../../chunk-5NBS5LAW.js";
import "../../../chunk-LY6UIYOG.js";
import "../../../chunk-7YJH7P7B.js";
import "../../../chunk-FAFQQMD4.js";
import {
  readFile
} from "../../../chunk-4IXC46ZA.js";
import "../../../chunk-LUGC3D2G.js";
import "../../../chunk-FQIRJADJ.js";
import "../../../chunk-FBB6KUZG.js";
import "../../../chunk-PCQFWMMD.js";
import "../../../chunk-SZQX5J7X.js";
import {
  require_lib
} from "../../../chunk-67MDUPX5.js";
import "../../../chunk-PBFWVFAQ.js";
import "../../../chunk-FWGRGI4R.js";
import "../../../chunk-BE73NZXB.js";
import "../../../chunk-VQZ2SHRY.js";
import "../../../chunk-6SL33DOA.js";
import "../../../chunk-WCNR75S2.js";
import {
  cwd,
  resolvePath
} from "../../../chunk-MHMFNQE3.js";
import "../../../chunk-3PIGYQSE.js";
import "../../../chunk-BZJX2NV6.js";
import "../../../chunk-RQ24VTW4.js";
import "../../../chunk-3JLUTHGR.js";
import "../../../chunk-GXPKATXW.js";
import {
  __toESM,
  init_cjs_shims
} from "../../../chunk-M63RTPGR.js";

// src/cli/commands/demo/index.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var Demo = class _Demo extends base_command_default {
  static {
    this.description = "Demo a command design defined in a file";
  }
  static {
    this.hidden = true;
  }
  static {
    this.flags = {
      file: import_core.Flags.string({
        hidden: false,
        description: "The path to the design file.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      })
    };
  }
  async run() {
    const { flags } = await this.parse(_Demo);
    const contents = await readFile(flags.file);
    const design = JSON.parse(contents);
    await demo(design);
  }
};
export {
  Demo as default
};
