// @shopify/cli/dist/cli/services/upgrade.test.js
import {
  upgrade
} from "../../chunk-3W2QB2OC.js";
import {
  mockAndCaptureOutput
} from "../../chunk-FFHJRSIP.js";
import {
  afterEach,
  beforeEach,
  describe,
  globalExpect,
  test,
  vi
} from "../../chunk-OXZ2SRCK.js";
import {
  platformAndArch
} from "../../chunk-OQBI57X3.js";
import {
  node_package_manager_exports
} from "../../chunk-LY6UIYOG.js";
import "../../chunk-7YJH7P7B.js";
import "../../chunk-FAFQQMD4.js";
import {
  AbortError,
  exec,
  inTemporaryDirectory,
  touchFile,
  writeFile
} from "../../chunk-4IXC46ZA.js";
import "../../chunk-LUGC3D2G.js";
import "../../chunk-FQIRJADJ.js";
import "../../chunk-FBB6KUZG.js";
import "../../chunk-PCQFWMMD.js";
import "../../chunk-SZQX5J7X.js";
import "../../chunk-67MDUPX5.js";
import "../../chunk-PBFWVFAQ.js";
import "../../chunk-FWGRGI4R.js";
import "../../chunk-BE73NZXB.js";
import "../../chunk-VQZ2SHRY.js";
import "../../chunk-6SL33DOA.js";
import "../../chunk-WCNR75S2.js";
import {
  joinPath,
  normalizePath
} from "../../chunk-MHMFNQE3.js";
import "../../chunk-3PIGYQSE.js";
import "../../chunk-BZJX2NV6.js";
import "../../chunk-RQ24VTW4.js";
import "../../chunk-3JLUTHGR.js";
import "../../chunk-GXPKATXW.js";
import {
  init_cjs_shims
} from "../../chunk-M63RTPGR.js";

// src/cli/services/upgrade.test.ts
init_cjs_shims();
var oldCliVersion = "3.0.0";
var currentCliVersion = "3.10.0";
vi.mock("@shopify/cli-kit/node/os", async () => {
  return {
    platformAndArch: vi.fn()
  };
});
vi.mock("@shopify/cli-kit/node/system");
beforeEach(async () => {
  vi.mocked(platformAndArch).mockReturnValue({ platform: "windows", arch: "amd64" });
});
afterEach(() => {
  mockAndCaptureOutput().clear();
});
describe("upgrade global CLI", () => {
  test("does not upgrade globally if the latest version is found", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      const outputMock = mockAndCaptureOutput();
      vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValue(void 0);
      await upgrade(tmpDir, currentCliVersion, { env: {} });
      globalExpect(outputMock.info()).toMatchInlineSnapshot(`
        "You're on the latest version, ${currentCliVersion}, no need to upgrade!"
      `);
    });
  });
  test("upgrades globally using npm if the latest version is not found", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      const outputMock = mockAndCaptureOutput();
      vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValue(currentCliVersion);
      await upgrade(tmpDir, oldCliVersion, { env: {} });
      globalExpect(vi.mocked(exec)).toHaveBeenCalledWith(
        "npm",
        ["install", "-g", "@shopify/cli@latest", "@shopify/theme@latest"],
        { stdio: "inherit" }
      );
      globalExpect(outputMock.info()).toMatchInlineSnapshot(`
        "Upgrading CLI from ${oldCliVersion} to ${currentCliVersion}...
Attempting to upgrade via \`npm install -g @shopify/cli@latest @shopify/theme@latest\`..."
      `);
      globalExpect(outputMock.success()).toMatchInlineSnapshot(`
        "Upgraded Shopify CLI to version ${currentCliVersion}"
      `);
    });
  });
  const homebrewPackageNames = ["shopify-cli", "shopify-cli@3"];
  homebrewPackageNames.forEach((homebrewPackageName) => {
    test("upgrades globally using Homebrew if the latest version is not found and the CLI was installed via Homebrew", async () => {
      await inTemporaryDirectory(async (tmpDir) => {
        vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValue(currentCliVersion);
        await globalExpect(async () => {
          await upgrade(tmpDir, oldCliVersion, { env: { SHOPIFY_HOMEBREW_FORMULA: homebrewPackageName } });
        }).rejects.toThrowError(AbortError);
      });
    });
  });
});
describe("upgrade local CLI", () => {
  test("throws an error if a valid app config file is missing", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      await Promise.all([
        writeFile(
          joinPath(tmpDir, "package.json"),
          JSON.stringify({ dependencies: { "@shopify/cli": currentCliVersion, "@shopify/app": currentCliVersion } })
        ),
        touchFile(joinPath(tmpDir, "shopify.wrongapp.toml"))
      ]);
      const outputMock = mockAndCaptureOutput();
      vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValue(void 0);
      await globalExpect(upgrade(tmpDir, currentCliVersion, { env: { npm_config_user_agent: "npm" } })).rejects.toBeInstanceOf(
        AbortError
      );
    });
  });
  test("does not upgrade locally if the latest version is found", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      await Promise.all([
        writeFile(
          joinPath(tmpDir, "package.json"),
          JSON.stringify({ dependencies: { "@shopify/cli": currentCliVersion, "@shopify/app": currentCliVersion } })
        ),
        touchFile(joinPath(tmpDir, "shopify.app.toml"))
      ]);
      const outputMock = mockAndCaptureOutput();
      vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValue(void 0);
      await upgrade(tmpDir, currentCliVersion, { env: { npm_config_user_agent: "npm" } });
      globalExpect(outputMock.info()).toMatchInlineSnapshot(`
        "You're on the latest version, ${currentCliVersion}, no need to upgrade!"
      `);
    });
  });
  test("upgrades locally if the latest version is not found", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      await Promise.all([
        writeFile(
          joinPath(tmpDir, "package.json"),
          JSON.stringify({ dependencies: { "@shopify/cli": oldCliVersion, "@shopify/app": oldCliVersion } })
        ),
        touchFile(joinPath(tmpDir, "shopify.app.toml"))
      ]);
      const outputMock = mockAndCaptureOutput();
      vi.spyOn(node_package_manager_exports, "checkForNewVersion").mockResolvedValueOnce(currentCliVersion);
      const addNPMDependenciesMock = vi.spyOn(node_package_manager_exports, "addNPMDependencies").mockResolvedValue(void 0);
      await upgrade(tmpDir, oldCliVersion, { env: {} });
      globalExpect(outputMock.info()).toMatchInlineSnapshot(`
        "Upgrading CLI from ${oldCliVersion} to ${currentCliVersion}..."
      `);
      globalExpect(addNPMDependenciesMock).toHaveBeenCalledWith([{ name: "@shopify/cli", version: "latest" }], {
        packageManager: "npm",
        type: "prod",
        directory: normalizePath(tmpDir),
        stdout: process.stdout,
        stderr: process.stderr,
        addToRootDirectory: false
      });
      globalExpect(outputMock.success()).toMatchInlineSnapshot(`
        "Upgraded Shopify CLI to version ${currentCliVersion}"
      `);
    });
  });
  test("upgrades locally if CLI is on latest version but APP isnt", async () => {
    await inTemporaryDirectory(async (tmpDir) => {
      await Promise.all([
        writeFile(
          joinPath(tmpDir, "package.json"),
          JSON.stringify({ dependencies: { "@shopify/cli": currentCliVersion, "@shopify/app": oldCliVersion } })
        ),
        touchFile(joinPath(tmpDir, "shopify.app.nondefault.toml"))
      ]);
      const outputMock = mockAndCaptureOutput();
      const checkMock = vi.spyOn(node_package_manager_exports, "checkForNewVersion");
      checkMock.mockResolvedValueOnce(void 0).mockResolvedValueOnce(currentCliVersion);
      const addNPMDependenciesMock = vi.spyOn(node_package_manager_exports, "addNPMDependencies").mockResolvedValue(void 0);
      await upgrade(tmpDir, oldCliVersion, { env: {} });
      globalExpect(outputMock.info()).toMatchInlineSnapshot(`
        "Upgrading CLI from ${oldCliVersion} to ${currentCliVersion}..."
      `);
      globalExpect(addNPMDependenciesMock).toHaveBeenCalledWith([{ name: "@shopify/cli", version: "latest" }], {
        packageManager: "npm",
        type: "prod",
        directory: normalizePath(tmpDir),
        stdout: process.stdout,
        stderr: process.stderr,
        addToRootDirectory: false
      });
      globalExpect(outputMock.success()).toMatchInlineSnapshot(`
        "Upgraded Shopify CLI to version ${currentCliVersion}"
      `);
    });
  });
});
