// @shopify/cli/dist/chunk-FWKRA64T.js
import {
  base_command_default
} from "./chunk-EDVQGTOL.js";
import {
  packageManagerFromUserAgent
} from "./chunk-LY6UIYOG.js";
import {
  currentProcessIsGlobal,
  inferPackageManagerForGlobalCLI,
  renderInfo
} from "./chunk-4IXC46ZA.js";
import {
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// src/cli/commands/upgrade.ts
init_cjs_shims();
var Upgrade = class extends base_command_default {
  static {
    this.summary = "Shows details on how to upgrade Shopify CLI.";
  }
  static {
    this.descriptionWithMarkdown = "Shows details on how to upgrade Shopify CLI.";
  }
  static {
    this.description = this.descriptionWithoutMarkdown();
  }
  async run() {
    const isGlobal = currentProcessIsGlobal();
    let packageManager = packageManagerFromUserAgent() ?? inferPackageManagerForGlobalCLI();
    if (packageManager === "unknown")
      packageManager = "npm";
    let installCommand = `${packageManager} i ${isGlobal ? "-g " : ""}@shopify/cli@latest`;
    if (packageManager === "yarn") {
      installCommand = `yarn ${isGlobal ? "global " : ""}add @shopify/cli@latest`;
    }
    renderInfo({
      body: [`To upgrade Shopify CLI use your package manager.
`, `Example:`, { command: installCommand }]
    });
  }
};

export {
  Upgrade
};
