// @shopify/cli/dist/custom-oclif-loader-KLTAWG4B-VOMMA3PO.js
import {
  fileExistsSync
} from "./chunk-EFL5CN6M.js";
import {
  cwd,
  joinPath
} from "./chunk-UV4APOMB.js";
import {
  require_lib2
} from "./chunk-74JT4RJX.js";
import "./chunk-3LWRRBH5.js";
import "./chunk-ZZGXXS5X.js";
import "./chunk-J4I3URIN.js";
import "./chunk-RTZHOWFC.js";
import {
  __toESM,
  init_cjs_shims as init_cjs_shims2
} from "./chunk-ZLGRKPW3.js";
import {
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// ../create-app/dist/custom-oclif-loader-KLTAWG4B.js
init_cjs_shims();
init_cjs_shims2();
var import_core = __toESM(require_lib2(), 1);
var ShopifyConfig = class extends import_core.Config {
  constructor(options) {
    const path = sniffForPath() ?? cwd();
    if (fileExistsSync(joinPath(`${path}`, "package.json"))) {
      options.pluginAdditions = {
        core: ["@shopify/cli-hydrogen"],
        path
      };
    }
    super(options);
    this["determinePriority"] = this.customPriority;
  }
  customPriority(commands) {
    const oclifPlugins = this.pjson.oclif?.plugins ?? [];
    const commandPlugins = commands.sort((aCommand, bCommand) => {
      const pluginAliasA = aCommand.pluginAlias ?? "A-Cannot-Find-This";
      const pluginAliasB = bCommand.pluginAlias ?? "B-Cannot-Find-This";
      const aIndex = oclifPlugins.indexOf(pluginAliasA);
      const bIndex = oclifPlugins.indexOf(pluginAliasB);
      if (aCommand.pluginType === "core" && bCommand.pluginAlias === "@shopify/cli-hydrogen") {
        return 1;
      }
      if (aCommand.pluginAlias === "@shopify/cli-hydrogen" && bCommand.pluginType === "core") {
        return -1;
      }
      if (aCommand.pluginType === "core" && bCommand.pluginType === "core") {
        return aIndex - bIndex;
      }
      if (bCommand.pluginType === "core" && aCommand.pluginType !== "core") {
        return 1;
      }
      if (aCommand.pluginType === "core" && bCommand.pluginType !== "core") {
        return -1;
      }
      if (aCommand.pluginType === "jit" && bCommand.pluginType !== "jit") {
        return 1;
      }
      if (bCommand.pluginType === "jit" && aCommand.pluginType !== "jit") {
        return -1;
      }
      return 0;
    });
    return commandPlugins[0];
  }
};
function sniffForPath() {
  const pathFlagIndex = process.argv.indexOf("--path");
  if (pathFlagIndex === -1)
    return;
  const pathFlag = process.argv[pathFlagIndex + 1];
  if (!pathFlag || pathFlag.startsWith("-"))
    return;
  return pathFlag;
}
export {
  ShopifyConfig,
  sniffForPath
};
