// @shopify/cli/dist/local-3LWDOA7J-UFWE2V6L.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
} from "./chunk-EFL5CN6M.js";
import "./chunk-UV4APOMB.js";
import "./chunk-74JT4RJX.js";
import "./chunk-3LWRRBH5.js";
import "./chunk-ZZGXXS5X.js";
import "./chunk-J4I3URIN.js";
import "./chunk-RTZHOWFC.js";
import "./chunk-ZLGRKPW3.js";
import {
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// ../create-app/dist/local-3LWDOA7J.js
init_cjs_shims();
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
};
