// @shopify/cli/dist/node-package-manager-TFY2ROCP-DDJAU44T.js
import {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  writePackageJSON,
  yarnLockfile
} from "./chunk-XRFOYWYI.js";
import "./chunk-EFL5CN6M.js";
import "./chunk-UV4APOMB.js";
import "./chunk-74JT4RJX.js";
import "./chunk-3LWRRBH5.js";
import "./chunk-ZZGXXS5X.js";
import "./chunk-J4I3URIN.js";
import "./chunk-RTZHOWFC.js";
import "./chunk-ZLGRKPW3.js";
import {
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// ../create-app/dist/node-package-manager-TFY2ROCP.js
init_cjs_shims();
export {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  writePackageJSON,
  yarnLockfile
};
